include/cliffordsynthesis/Results.hpp

```diff
@@ -0,0 +1,157 @@
+            os << R"("verbosity":")" << verbose << "\"," << std::endl;
+            os << R"("choose_best":")" << chooseBest << "\"," << std::endl;
+            os << R"("result":")" << logicbase::toString(result) << "\"," << std::endl;
+            os << R"("strategy":")" << toString(strategy) << "\"," << std::endl;
+            os << R"("target":")" << toString(target) << "\"," << std::endl;
+            os << R"("method":")" << toString(method) << "\"," << std::endl;
+            os << R"("qubits":")" << std::to_string(nqubits) << "\"," << std::endl;
+            os << R"("initial_timesteps":")" << std::to_string(initialTimesteps)
+               << "\"," << std::endl;
+            os << R"("gate_count":")" << std::to_string(gateCount) << "\","
+               << std::endl;
+            os << R"("depth":")" << std::to_string(depth) << "\"," << std::endl;
+            os << R"("fidelity":")" << std::to_string(fidelity) << "\"," << std::endl;
+            os << R"("sat":")" << (sat ? "SAT" : "UNSAT") << "\"," << std::endl;
+            os << R"("total_seconds":")" << std::to_string(totalSeconds) << "\","
+               << std::endl;
+            os << R"("resultCircuit":")";
+            os << resultStringCircuit << "\"," << std::endl;
+            os << "\"resultTableaus\":[" << std::endl;
+            bool skipfirst = true;
+            for (const auto& tableau: resultTableaus) {
+                if (!skipfirst) {
+                    os << "," << std::endl;
+                }
+                os << "\"";
+                os << escapeChars(tableau.toString(), "\"");
+                os << "\"";
+                skipfirst = false;
+            }
+            os << "]," << std::endl;
+            std::stringstream strings;
+            Architecture::printCouplingMap(resultCM, strings);
+            os << R"("CouplingMap":")" << strings.str() << "\","
+               << std::endl;
+            os << "\"singleFidelity\":[";
+            skipfirst = true;
+            for (const auto& f: singleFidelity) {
+                if (!skipfirst) {
+                    os << ",";
+                }
+                os << "\"" << std::to_string(f) << "\"";
+                skipfirst = false;
+            }
+            os << "]," << std::endl;
+            os << "\"doubleFidelity\":[";
+            skipfirst = true;
+            for (const auto& f: doubleFidelity) {
+                if (!skipfirst) {
+                    os << ",";
+                }
+                os << "[";
+                bool skipfirst2 = true;
+                for (const auto& f2: f) {
+                    if (!skipfirst2) {
+                        os << ",";
+                    }
+                    os << "\"" << std::to_string(f2) << "\"";
+                    skipfirst2 = false;
+                }
+                os << "]";
+                skipfirst = false;
+            }
+            os << "]" << std::endl;
+            os << "}}" << std::endl;
```

Comment:
avoid the redundancy of manually building a JSON entry here. Just write the JSON function properly with the desired output and use the corresponding `dump` function of the JSON object.